cli/lib/kontena/cli/stacks/install_command.rb

```diff
@@ -16,21 +17,59 @@ class InstallCommand < Kontena::Command
     include Common::StackValuesToOption
     include Common::StackValuesFromOption
 
+    option '--parent-name', '[PARENT_NAME]', "Set parent stack name", hidden: true
+    option '--skip-dependencies', :flag, "Do not install any stack dependencies"
+
     requires_current_master
     requires_current_master_token
 
     def execute
-      stack = stack_read_and_dump(filename, name: name, values: values)
+      install_dependencies unless skip_dependencies?
+
+      set_env_variables(stack_name, current_grid)
+
+      hint_on_validation_notifications(stack['notifications']) unless stack['notifications'].empty?
+      abort_on_validation_errors(stack['errors']) unless stack['errors'].empty?
+
+      dump_variables if values_to
+
+      create_stack
+      deploy_stack if deploy?
+    end
+
+    def install_dependencies
+      dependencies = loader.dependencies
+      return if dependencies.nil?
+      dependencies.each do |dependency|
+        target_name = "#{stack_name}-#{dependency['name']}"
+        caret "Installing dependency #{pastel.cyan(dependency[:stack])} as #{pastel.cyan(target_name)}"
+        cmd = ['stack', 'install', '-n', target_name, '--parent-name', stack_name]
+
+        dependency['variables'].merge(dependency_values_from_options(dependency['name'])).each do |key, value|
+          next if key == 'PARENT_STACK'
+          cmd.concat ['-v', "#{key}=#{value}"]
+        end
+
+        cmd << '--no-deploy' unless deploy?
+
+        cmd << dependency['stack']
+        Kontena.run!(cmd)
+      end
+    end
 
-      stack['name'] = name if name
+    def create_stack
       spinner "Creating stack #{pastel.cyan(stack['name'])} " do
-        create_stack(stack)
+        stack.delete('errors')
+        stack.delete('notifications')
+        stack['services'].each do |svc|
+          svc['env'] = svc.delete('environment')
```

Comment:
Wut no, just revert the `env` name in the `ServiceGenerator`?

Also applies to the stack upgrade `PUT` etc.